query/query.go

```diff
@@ -30,12 +30,52 @@ var _ = log.Println
 type Q interface {
 	String() string
 }
-type Visibility struct {
-	Value string
+
+const (
+	Public     uint8 = 1
+	Private          = 2
+	Forks            = 1 << 2
+	NoForks          = 2 << 2
+	Archived         = 1 << 4
+	NoArchived       = 2 << 4
+)
+
+var flagNames = map[uint8]string{
+	Public:     "Public",
+	Private:    "Private",
+	Forks:      "Forks",
+	NoForks:    "NoForks",
+	Archived:   "Archived",
+	NoArchived: "NoArchived",
+}
+
+// NewRawConfig construct a RawConfig query atom. The default value flag=0 will
+// filter for all public repositories which are neither forks nor archived. Not
+// providing a flag is equivalent to ignoring the dimension. For example,
+// Forks|Archived will filter for public and private repositories which are both,
+// forks and archived.
+func NewRawConfig(flag uint8) *RawConfig {
+	if flag == 0 {
+		return &RawConfig{Encoded: Public | NoForks | NoArchived}
+	}
+	return &RawConfig{flag}
 }
 
-func (v *Visibility) String() string {
-	return fmt.Sprintf("visibility:%s", v.Value)
+// RawConfig filters for repositories based on their encoded RawConfig map.
+// Always use the constructor NewRawConfig instead of creating RawConfig
+// manually.
+type RawConfig struct {
+	Encoded uint8
```

Comment:
this is part of the public API. You might want to use a larger type to give more space in the future without breaking the API at all. I recommend uint64. You can keep it uint8 on indexData until it grows.